testsuite/shared/src/main/java/org/jboss/as/test/shared/util/LoggingUtil.java

```diff
@@ -0,0 +1,94 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.shared.util;
+
+import java.io.BufferedReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.dmr.ModelNode;
+
+/**
+ * Utility class for logging repetitive tasks
+ * @author tmiyar
+ *
+ */
+public class LoggingUtil {
+
+    /**
+     * Will return the full log path for the handler in the name parameter,
+     * to be used when using @RunAsClient annotation
+     * Depending on how you run it you might need some permissions:
+     * new PropertyPermission("node0", "read"),
+     * new RemotingPermission("connect"),
+     * new SocketPermission(Utils.getDefaultHost(true), "accept,connect,listen,resolve"),
+     * new RuntimePermission("getClassLoader")
+     * @param managementClient
+     * @param name of handler
+     * @param handlerType i.e. periodic-rotating-file-handler
+     * @return
+     * @throws Exception
+     */
+    public static Path getLogPath(ManagementClient managementClient, String handlerType, String name) throws Exception {
+        final ModelNode address = Operations.createAddress("subsystem", "logging", handlerType, name);
+        final ModelNode op = Operations.createOperation("resolve-path", address);
+        final ModelNode result = managementClient.getControllerClient().execute(op);
+        if (!Operations.isSuccessfulOutcome(result)) {
+            throw new Exception("Can't get log file");
+        }
+        return Paths.get(Operations.readResult(result).asString());
+    }
+
+    /**
+     * Will return the full log path for the given log file relative to the the jboss.server.log.dir.
+     * Meant for use by test code that runs in the server VM. Tests that use this should add
+     * the following permission to allow the call to succeed in a testsuite run with the security manager enabled:
+     * new PropertyPermission("jboss.server.log.dir", "read")
+     *
+     * @param logFile name of the log file, relative to the server log directory
+     * @return the path
+     */
+    public static Path getInServerLogPath(String logFile) {
+        return Paths.get(System.getProperty("jboss.server.log.dir")).resolve(logFile);
+    }
+
+    public static boolean hasLogMessage(String logFileName, String logMessage) throws Exception {
+
+        BufferedReader fileReader = null;
+        boolean found = false;
+        Path logPath = LoggingUtil.getInServerLogPath(logFileName);
+
+        fileReader = Files.newBufferedReader(logPath);
```

Comment:
We should probably just use a try with resources here. At a minimum we'd need the close in a finally block. Something like
```
    public static boolean hasLogMessage(String logFileName, String logMessage) throws Exception {

        boolean found = false;
        Path logPath = LoggingUtil.getInServerLogPath(logFileName);

        try (BufferedReader fileReader = Files.newBufferedReader(logPath)) {

            String line = "";
            while ((line = fileReader.readLine()) != null) {
                if (line.contains(logMessage)) {
                    found = true;
                    break;
                }
            }
        }

        return found;
    }
```